test/oban/testing_test.exs

```diff
@@ -320,6 +325,20 @@ defmodule Oban.TestingTest do
         refute_received {:ok, 1}
       end)
     end
+
+    test "with perform_job" do
```

Comment:
These are mostly guarding against regressions without serving a lot of inherent value to a reader, IMO. Happy to take suggestions.